src/LiveQuery/ParseLiveQueryServer.js

```diff
@@ -633,8 +655,10 @@ class ParseLiveQueryServer {
       };
       const trigger = getTrigger('@Connect', 'beforeConnect', Parse.applicationId);
       if (trigger) {
-        const auth = await this.getAuthForSessionToken(req.sessionToken);
-        req.user = auth.user;
+        const auth = await this.getAuthFromClient(client, request.requestId, req.sessionToken);
+        if (auth && auth.user) {
+          req.user = auth.user;
+        }
         await runTrigger(trigger, `beforeConnect.@Connect`, req, auth);
```

Comment:
Is it even possible that a user with invalid or missing auth connects and triggers the beforeConnect event?

Reply:
Yes - same as before, if you want to use LQ without requiring people to login

Reply:
Got it, I think this looks fine now